From 260f5a2dd17170e289b38268beee04f4301ce40a Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Thu, 4 Oct 2018 14:24:51 +0900
Subject: [PATCH 17/27] nsis: Use DropLists instead of RadioButtons

The _vimrc setting screen was too noisy.
Also adjust wording.
---
 nsis/gvim.nsi             | 136 +++++++++++++++++---------------------
 nsis/lang/dutch.nsi       |  30 +++++----
 nsis/lang/english.nsi     |  30 +++++----
 nsis/lang/german.nsi      |  30 +++++----
 nsis/lang/italian.nsi     |  30 +++++----
 nsis/lang/japanese.nsi    |  22 +++---
 nsis/lang/simpchinese.nsi |  30 +++++----
 nsis/lang/tradchinese.nsi |  30 +++++----
 8 files changed, 181 insertions(+), 157 deletions(-)

diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index eea123cdd..a8a095372 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -153,15 +153,9 @@ Page custom SetCustom ValidateCustom
 
 # Global variables
 Var vim_dialog
-Var vim_nsd_compat_1
-Var vim_nsd_compat_2
-Var vim_nsd_compat_3
-Var vim_nsd_compat_4
-Var vim_nsd_keymap_1
-Var vim_nsd_keymap_2
-Var vim_nsd_mouse_1
-Var vim_nsd_mouse_2
-Var vim_nsd_mouse_3
+Var vim_nsd_compat
+Var vim_nsd_keymap
+Var vim_nsd_mouse
 Var vim_compat_stat
 Var vim_keymap_stat
 Var vim_mouse_stat
@@ -705,90 +699,87 @@ Function SetCustom
 
 
 	# 1st group - Compatibility
-	${NSD_CreateGroupBox} 0 0 100% 28% $(str_msg_compat_title)
+	${NSD_CreateGroupBox} 0 0 100% 24% $(str_msg_compat_title)
 	Pop $3
 
-	${NSD_CreateRadioButton} 5% 8% 40% 8% $(str_msg_compat_vi)
-	Pop $vim_nsd_compat_1
-	${NSD_AddStyle} $vim_nsd_compat_1 ${WS_GROUP}
-
-	${NSD_CreateRadioButton} 50% 8% 40% 8% $(str_msg_compat_vim)
-	Pop $vim_nsd_compat_2
-
-	${NSD_CreateRadioButton} 5% 17% 40% 8% $(str_msg_compat_defaults)
-	Pop $vim_nsd_compat_3
-
-	${NSD_CreateRadioButton} 50% 17% 40% 8% $(str_msg_compat_all)
-	Pop $vim_nsd_compat_4
-
-	# Default button
+	${NSD_CreateLabel} 5% 10% 35% 8% $(str_msg_compat_desc)
+	Pop $3
+	${NSD_CreateDropList} 48% 10% 45% 8% ""
+	Pop $vim_nsd_compat
+	${NSD_CB_AddString} $vim_nsd_compat $(str_msg_compat_vi)
+	${NSD_CB_AddString} $vim_nsd_compat $(str_msg_compat_vim)
+	${NSD_CB_AddString} $vim_nsd_compat $(str_msg_compat_defaults)
+	${NSD_CB_AddString} $vim_nsd_compat $(str_msg_compat_all)
+
+	# Default selection
 	${If} $vim_compat_stat == ""
 	  ReadRegStr $3 HKLM "${UNINST_REG_KEY_VIM}" "compat"
 	${Else}
 	  StrCpy $3 $vim_compat_stat
 	${EndIf}
 	${If} $3 == "defaults"
-	  ${NSD_SetState} $vim_nsd_compat_3 ${BST_CHECKED}
+	  StrCpy $4 2
 	${ElseIf} $3 == "vim"
-	  ${NSD_SetState} $vim_nsd_compat_2 ${BST_CHECKED}
+	  StrCpy $4 1
 	${ElseIf} $3 == "vi"
-	  ${NSD_SetState} $vim_nsd_compat_1 ${BST_CHECKED}
-	${Else} # defualt
-	  ${NSD_SetState} $vim_nsd_compat_4 ${BST_CHECKED}
+	  StrCpy $4 0
+	${Else} # default
+	  StrCpy $4 3
 	${EndIf}
+	${NSD_CB_SetSelectionIndex} $vim_nsd_compat $4
 
 
 	# 2nd group - Key remapping
-	${NSD_CreateGroupBox} 0 31% 100% 28% $(str_msg_keymap_title)
+	${NSD_CreateGroupBox} 0 27% 100% 33% $(str_msg_keymap_title)
 	Pop $3
 
-	${NSD_CreateRadioButton} 5% 39% 90% 8% $(str_msg_keymap_default)
-	Pop $vim_nsd_keymap_1
-	${NSD_AddStyle} $vim_nsd_keymap_1 ${WS_GROUP}
-
-	${NSD_CreateRadioButton} 5% 48% 90% 8% $(str_msg_keymap_windows)
-	Pop $vim_nsd_keymap_2
+	${NSD_CreateLabel} 5% 37% 90% 8% $(str_msg_keymap_desc)
+	Pop $3
+	${NSD_CreateDropList} 48% 47% 45% 8% ""
+	Pop $vim_nsd_keymap
+	${NSD_CB_AddString} $vim_nsd_keymap $(str_msg_keymap_default)
+	${NSD_CB_AddString} $vim_nsd_keymap $(str_msg_keymap_windows)
 
-	# Default button
+	# Default selection
 	${If} $vim_keymap_stat == ""
 	  ReadRegStr $3 HKLM "${UNINST_REG_KEY_VIM}" "keyremap"
 	${Else}
 	  StrCpy $3 $vim_keymap_stat
 	${EndIf}
 	${If} $3 == "windows"
-	  ${NSD_SetState} $vim_nsd_keymap_2 ${BST_CHECKED}
+	  StrCpy $4 1
 	${Else} # default
-	  ${NSD_SetState} $vim_nsd_keymap_1 ${BST_CHECKED}
+	  StrCpy $4 0
 	${EndIf}
+	${NSD_CB_SetSelectionIndex} $vim_nsd_keymap $4
 
 
 	# 3rd group - Mouse behavior
-	${NSD_CreateGroupBox} 0 62% 100% 37% $(str_msg_mouse_title)
+	${NSD_CreateGroupBox} 0 63% 100% 36% $(str_msg_mouse_title)
 	Pop $3
 
-	${NSD_CreateRadioButton} 5% 70% 90% 8% $(str_msg_mouse_default)
-	Pop $vim_nsd_mouse_1
-	${NSD_AddStyle} $vim_nsd_mouse_1 ${WS_GROUP}
-
-	${NSD_CreateRadioButton} 5% 79% 90% 8% $(str_msg_mouse_windows)
-	Pop $vim_nsd_mouse_2
-
-	${NSD_CreateRadioButton} 5% 88% 90% 8% $(str_msg_mouse_unix)
-	Pop $vim_nsd_mouse_3
+	${NSD_CreateLabel} 5% 73% 90% 8% $(str_msg_mouse_desc)
+	Pop $3
+	${NSD_CreateDropList} 23% 82% 70% 8% ""
+	Pop $vim_nsd_mouse
+	${NSD_CB_AddString} $vim_nsd_mouse $(str_msg_mouse_default)
+	${NSD_CB_AddString} $vim_nsd_mouse $(str_msg_mouse_windows)
+	${NSD_CB_AddString} $vim_nsd_mouse $(str_msg_mouse_unix)
 
-	# Default button
+	# Default selection
 	${If} $vim_mouse_stat == ""
 	  ReadRegStr $3 HKLM "${UNINST_REG_KEY_VIM}" "mouse"
 	${Else}
 	  StrCpy $3 $vim_mouse_stat
 	${EndIf}
 	${If} $3 == "xterm"
-	  ${NSD_SetState} $vim_nsd_mouse_3 ${BST_CHECKED}
+	  StrCpy $4 2
 	${ElseIf} $3 == "windows"
-	  ${NSD_SetState} $vim_nsd_mouse_2 ${BST_CHECKED}
-	${Else} # defualt
-	  ${NSD_SetState} $vim_nsd_mouse_1 ${BST_CHECKED}
+	  StrCpy $4 1
+	${Else} # default
+	  StrCpy $4 0
 	${EndIf}
+	${NSD_CB_SetSelectionIndex} $vim_nsd_mouse $4
 
 	${If} ${RunningX64}
 	  SetRegView lastused
@@ -798,40 +789,31 @@ Function SetCustom
 FunctionEnd
 
 Function ValidateCustom
-	${NSD_GetState} $vim_nsd_compat_1 $3
-	${If} $3 == ${BST_CHECKED}
+	${NSD_CB_GetSelectionIndex} $vim_nsd_compat $3
+	${If} $3 = 0
 	  StrCpy $vim_compat_stat "vi"
+	${ElseIf} $3 = 1
+	  StrCpy $vim_compat_stat "vim"
+	${ElseIf} $3 = 2
+	  StrCpy $vim_compat_stat "defaults"
 	${Else}
-	  ${NSD_GetState} $vim_nsd_compat_2 $3
-	  ${If} $3 == ${BST_CHECKED}
-	    StrCpy $vim_compat_stat "vim"
-	  ${Else}
-	    ${NSD_GetState} $vim_nsd_compat_3 $3
-	    ${If} $3 == ${BST_CHECKED}
-	      StrCpy $vim_compat_stat "defaults"
-	    ${Else}
-	      StrCpy $vim_compat_stat "all"
-	    ${EndIf}
-	  ${EndIf}
+	  StrCpy $vim_compat_stat "all"
 	${EndIf}
 
-	${NSD_GetState} $vim_nsd_keymap_1 $3
-	${If} $3 == ${BST_CHECKED}
+	${NSD_CB_GetSelectionIndex} $vim_nsd_keymap $3
+	${If} $3 = 0
 	  StrCpy $vim_keymap_stat "default"
 	${Else}
 	  StrCpy $vim_keymap_stat "windows"
 	${EndIf}
 
-	${NSD_GetState} $vim_nsd_mouse_1 $3
-	${If} $3 == ${BST_CHECKED}
+	${NSD_CB_GetSelectionIndex} $vim_nsd_mouse $3
+	${If} $3 = 0
 	  StrCpy $vim_mouse_stat "default"
+	${ElseIf} $3 = 1
+	  StrCpy $vim_mouse_stat "windows"
 	${Else}
-	  ${NSD_GetState} $vim_nsd_mouse_2 $3
-	  ${If} $3 == ${BST_CHECKED}
-	    StrCpy $vim_mouse_stat "windows"
-	  ${Else}
-	    StrCpy $vim_mouse_stat "xterm"
-	  ${EndIf}
+	  StrCpy $vim_mouse_stat "xterm"
 	${EndIf}
 FunctionEnd
 
diff --git a/nsis/lang/dutch.nsi b/nsis/lang/dutch.nsi
index c8b6d2c68..62fceb736 100644
--- a/nsis/lang/dutch.nsi
+++ b/nsis/lang/dutch.nsi
@@ -229,31 +229,37 @@ LangString str_msg_rm_root_fail  ${LANG_DUTCH} \
 LangString str_vimrc_page_title    ${LANG_DUTCH} \
     "Choose _vimrc settings"
 LangString str_vimrc_page_subtitle ${LANG_DUTCH} \
-    "Choose the default _vimrc settings."
+    "Choose the settings for enhancement, keyboard and mouse."
 
 LangString str_msg_compat_title    ${LANG_DUTCH} \
-    " Vi Compatibility "
+    " Vi / Vim behavior "
+LangString str_msg_compat_desc     ${LANG_DUTCH} \
+    "&Compatibility and enhancements"
 LangString str_msg_compat_vi       ${LANG_DUTCH} \
-    " Vi compatible"
+    "Vi compatible"
 LangString str_msg_compat_vim      ${LANG_DUTCH} \
-    " Vim original"
+    "Vim original"
 LangString str_msg_compat_defaults ${LANG_DUTCH} \
-    " Some enhancements"
+    "Vim with some enhancements"
 LangString str_msg_compat_all      ${LANG_DUTCH} \
-    " All enhancements (Default)"
+    "Vim with all enhancements (Default)"
 
 LangString str_msg_keymap_title   ${LANG_DUTCH} \
-    " Key remapping for Windows behavior "
+    " Mappings "
+LangString str_msg_keymap_desc    ${LANG_DUTCH} \
+    "&Remap a few keys (Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F, etc)"
 LangString str_msg_keymap_default ${LANG_DUTCH} \
     " Do not remap keys (Default)"
 LangString str_msg_keymap_windows ${LANG_DUTCH} \
-    " Remap a few keys (<C-V>, <C-C>, <C-A>, <C-S>, <C-F>, etc)"
+    " Remap a few keys"
 
 LangString str_msg_mouse_title   ${LANG_DUTCH} \
-    " Mouse behavior "
+    " Mouse "
+LangString str_msg_mouse_desc    ${LANG_DUTCH} \
+    "&Behavior of right and left buttons"
 LangString str_msg_mouse_default ${LANG_DUTCH} \
-    " Default:    Right button has a popup menu, left button starts visual mode"
+    "Right: popup menu, Left: visual mode (Default)"
 LangString str_msg_mouse_windows ${LANG_DUTCH} \
-    " Windows: Right button has a popup menu, left button starts select mode"
+    "Right: popup menu, Left: select mode (Windows)"
 LangString str_msg_mouse_unix    ${LANG_DUTCH} \
-    " Unix:        Right button extends selection, left button starts visual mode"
+    "Right: extends selection, Left: visual mode (Unix)"
diff --git a/nsis/lang/english.nsi b/nsis/lang/english.nsi
index 700a88fb8..8a14bcbd3 100644
--- a/nsis/lang/english.nsi
+++ b/nsis/lang/english.nsi
@@ -238,31 +238,37 @@ LangString str_msg_rm_root_fail  ${LANG_ENGLISH} \
 LangString str_vimrc_page_title    ${LANG_ENGLISH} \
     "Choose _vimrc settings"
 LangString str_vimrc_page_subtitle ${LANG_ENGLISH} \
-    "Choose the default _vimrc settings."
+    "Choose the settings for enhancement, keyboard and mouse."
 
 LangString str_msg_compat_title    ${LANG_ENGLISH} \
-    " Vi Compatibility "
+    " Vi / Vim behavior "
+LangString str_msg_compat_desc     ${LANG_ENGLISH} \
+    "&Compatibility and enhancements"
 LangString str_msg_compat_vi       ${LANG_ENGLISH} \
-    " Vi compatible"
+    "Vi compatible"
 LangString str_msg_compat_vim      ${LANG_ENGLISH} \
-    " Vim original"
+    "Vim original"
 LangString str_msg_compat_defaults ${LANG_ENGLISH} \
-    " Some enhancements"
+    "Vim with some enhancements"
 LangString str_msg_compat_all      ${LANG_ENGLISH} \
-    " All enhancements (Default)"
+    "Vim with all enhancements (Default)"
 
 LangString str_msg_keymap_title   ${LANG_ENGLISH} \
-    " Key remapping for Windows behavior "
+    " Mappings "
+LangString str_msg_keymap_desc    ${LANG_ENGLISH} \
+    "&Remap a few keys (Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F, etc)"
 LangString str_msg_keymap_default ${LANG_ENGLISH} \
     " Do not remap keys (Default)"
 LangString str_msg_keymap_windows ${LANG_ENGLISH} \
-    " Remap a few keys (<C-V>, <C-C>, <C-A>, <C-S>, <C-F>, etc)"
+    " Remap a few keys"
 
 LangString str_msg_mouse_title   ${LANG_ENGLISH} \
-    " Mouse behavior "
+    " Mouse "
+LangString str_msg_mouse_desc    ${LANG_ENGLISH} \
+    "&Behavior of right and left buttons"
 LangString str_msg_mouse_default ${LANG_ENGLISH} \
-    " Default:    Right button has a popup menu, left button starts visual mode"
+    "Right: popup menu, Left: visual mode (Default)"
 LangString str_msg_mouse_windows ${LANG_ENGLISH} \
-    " Windows: Right button has a popup menu, left button starts select mode"
+    "Right: popup menu, Left: select mode (Windows)"
 LangString str_msg_mouse_unix    ${LANG_ENGLISH} \
-    " Unix:        Right button extends selection, left button starts visual mode"
+    "Right: extends selection, Left: visual mode (Unix)"
diff --git a/nsis/lang/german.nsi b/nsis/lang/german.nsi
index 895e458ba..b9198defe 100644
--- a/nsis/lang/german.nsi
+++ b/nsis/lang/german.nsi
@@ -229,31 +229,37 @@ LangString str_msg_rm_root_fail  ${LANG_GERMAN} \
 LangString str_vimrc_page_title    ${LANG_GERMAN} \
     "Choose _vimrc settings"
 LangString str_vimrc_page_subtitle ${LANG_GERMAN} \
-    "Choose the default _vimrc settings."
+    "Choose the settings for enhancement, keyboard and mouse."
 
 LangString str_msg_compat_title    ${LANG_GERMAN} \
-    " Vi Compatibility "
+    " Vi / Vim behavior "
+LangString str_msg_compat_desc     ${LANG_GERMAN} \
+    "&Compatibility and enhancements"
 LangString str_msg_compat_vi       ${LANG_GERMAN} \
-    " Vi compatible"
+    "Vi compatible"
 LangString str_msg_compat_vim      ${LANG_GERMAN} \
-    " Vim original"
+    "Vim original"
 LangString str_msg_compat_defaults ${LANG_GERMAN} \
-    " Some enhancements"
+    "Vim with some enhancements"
 LangString str_msg_compat_all      ${LANG_GERMAN} \
-    " All enhancements (Default)"
+    "Vim with all enhancements (Default)"
 
 LangString str_msg_keymap_title   ${LANG_GERMAN} \
-    " Key remapping for Windows behavior "
+    " Mappings "
+LangString str_msg_keymap_desc    ${LANG_GERMAN} \
+    "&Remap a few keys (Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F, etc)"
 LangString str_msg_keymap_default ${LANG_GERMAN} \
     " Do not remap keys (Default)"
 LangString str_msg_keymap_windows ${LANG_GERMAN} \
-    " Remap a few keys (<C-V>, <C-C>, <C-A>, <C-S>, <C-F>, etc)"
+    " Remap a few keys"
 
 LangString str_msg_mouse_title   ${LANG_GERMAN} \
-    " Mouse behavior "
+    " Mouse "
+LangString str_msg_mouse_desc    ${LANG_GERMAN} \
+    "&Behavior of right and left buttons"
 LangString str_msg_mouse_default ${LANG_GERMAN} \
-    " Default:    Right button has a popup menu, left button starts visual mode"
+    "Right: popup menu, Left: visual mode (Default)"
 LangString str_msg_mouse_windows ${LANG_GERMAN} \
-    " Windows: Right button has a popup menu, left button starts select mode"
+    "Right: popup menu, Left: select mode (Windows)"
 LangString str_msg_mouse_unix    ${LANG_GERMAN} \
-    " Unix:        Right button extends selection, left button starts visual mode"
+    "Right: extends selection, Left: visual mode (Unix)"
diff --git a/nsis/lang/italian.nsi b/nsis/lang/italian.nsi
index 6f6e20bdc..7de925e6c 100644
--- a/nsis/lang/italian.nsi
+++ b/nsis/lang/italian.nsi
@@ -238,31 +238,37 @@ LangString str_msg_rm_root_fail  ${LANG_ITALIAN} \
 LangString str_vimrc_page_title    ${LANG_ITALIAN} \
     "Choose _vimrc settings"
 LangString str_vimrc_page_subtitle ${LANG_ITALIAN} \
-    "Choose the default _vimrc settings."
+    "Choose the settings for enhancement, keyboard and mouse."
 
 LangString str_msg_compat_title    ${LANG_ITALIAN} \
-    " Vi Compatibility "
+    " Vi / Vim behavior "
+LangString str_msg_compat_desc     ${LANG_ITALIAN} \
+    "&Compatibility and enhancements"
 LangString str_msg_compat_vi       ${LANG_ITALIAN} \
-    " Vi compatible"
+    "Vi compatible"
 LangString str_msg_compat_vim      ${LANG_ITALIAN} \
-    " Vim original"
+    "Vim original"
 LangString str_msg_compat_defaults ${LANG_ITALIAN} \
-    " Some enhancements"
+    "Vim with some enhancements"
 LangString str_msg_compat_all      ${LANG_ITALIAN} \
-    " All enhancements (Default)"
+    "Vim with all enhancements (Default)"
 
 LangString str_msg_keymap_title   ${LANG_ITALIAN} \
-    " Key remapping for Windows behavior "
+    " Mappings "
+LangString str_msg_keymap_desc    ${LANG_ITALIAN} \
+    "&Remap a few keys (Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F, etc)"
 LangString str_msg_keymap_default ${LANG_ITALIAN} \
     " Do not remap keys (Default)"
 LangString str_msg_keymap_windows ${LANG_ITALIAN} \
-    " Remap a few keys (<C-V>, <C-C>, <C-A>, <C-S>, <C-F>, etc)"
+    " Remap a few keys"
 
 LangString str_msg_mouse_title   ${LANG_ITALIAN} \
-    " Mouse behavior "
+    " Mouse "
+LangString str_msg_mouse_desc    ${LANG_ITALIAN} \
+    "&Behavior of right and left buttons"
 LangString str_msg_mouse_default ${LANG_ITALIAN} \
-    " Default:    Right button has a popup menu, left button starts visual mode"
+    "Right: popup menu, Left: visual mode (Default)"
 LangString str_msg_mouse_windows ${LANG_ITALIAN} \
-    " Windows: Right button has a popup menu, left button starts select mode"
+    "Right: popup menu, Left: select mode (Windows)"
 LangString str_msg_mouse_unix    ${LANG_ITALIAN} \
-    " Unix:        Right button extends selection, left button starts visual mode"
+    "Right: extends selection, Left: visual mode (Unix)"
diff --git a/nsis/lang/japanese.nsi b/nsis/lang/japanese.nsi
index 678fa80e5..3f1411b6f 100644
--- a/nsis/lang/japanese.nsi
+++ b/nsis/lang/japanese.nsi
@@ -234,10 +234,12 @@ LangString str_msg_rm_root_fail  ${LANG_JAPANESE} \
 LangString str_vimrc_page_title    ${LANG_JAPANESE} \
     "_vimrc の設定を選んでください"
 LangString str_vimrc_page_subtitle ${LANG_JAPANESE} \
-    "既定の _vimrc の設定を選んでください。"
+    "拡張やキーボード、マウスの設定を選んでください。"
 
 LangString str_msg_compat_title    ${LANG_JAPANESE} \
-    " Vi との互換性 "
+    " Vi / Vim の動作 "
+LangString str_msg_compat_desc     ${LANG_JAPANESE} \
+    "互換性と拡張(&C)"
 LangString str_msg_compat_vi       ${LANG_JAPANESE} \
     " Vi 互換"
 LangString str_msg_compat_vim      ${LANG_JAPANESE} \
@@ -248,17 +250,21 @@ LangString str_msg_compat_all      ${LANG_JAPANESE} \
     " Vim 独自と全ての拡張 (既定)"
 
 LangString str_msg_keymap_title   ${LANG_JAPANESE} \
-    " Windows用のキーのリマッピング "
+    " マッピング "
+LangString str_msg_keymap_desc    ${LANG_JAPANESE} \
+    "いくつかのキーをリマップする(&R) (例: Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F など)"
 LangString str_msg_keymap_default ${LANG_JAPANESE} \
     " リマップしない (既定)"
 LangString str_msg_keymap_windows ${LANG_JAPANESE} \
-    " いくつかのキーをリマップする (例: <C-V>, <C-C>, <C-A>, <C-S>, <C-F> など)"
+    " リマップする"
 
 LangString str_msg_mouse_title   ${LANG_JAPANESE} \
-    " マウスの動作 "
+    " マウス "
+LangString str_msg_mouse_desc    ${LANG_JAPANESE} \
+    "右ボタンと左ボタンの動作(&B)"
 LangString str_msg_mouse_default ${LANG_JAPANESE} \
-    " 既定:       右ボタンはポップアップメニュー、左ボタンはビジュアルモードを開始"
+    "右：ポップアップメニュー、左：ビジュアルモード (既定)"
 LangString str_msg_mouse_windows ${LANG_JAPANESE} \
-    " Windows: 右ボタンはポップアップメニュー、左ボタンは選択モードを開始"
+    "右：ポップアップメニュー、左：選択モード (Windows)"
 LangString str_msg_mouse_unix    ${LANG_JAPANESE} \
-    " Unix:       右ボタンは選択を拡張、左ボタンはビジュアルモードを開始"
+    "右：選択を拡張、左：ビジュアルモード (Unix)"
diff --git a/nsis/lang/simpchinese.nsi b/nsis/lang/simpchinese.nsi
index f835a635f..37b442214 100644
--- a/nsis/lang/simpchinese.nsi
+++ b/nsis/lang/simpchinese.nsi
@@ -225,31 +225,37 @@ LangString str_msg_rm_root_fail  ${LANG_SIMPCHINESE} \
 LangString str_vimrc_page_title    ${LANG_SIMPCHINESE} \
     "Choose _vimrc settings"
 LangString str_vimrc_page_subtitle ${LANG_SIMPCHINESE} \
-    "Choose the default _vimrc settings."
+    "Choose the settings for enhancement, keyboard and mouse."
 
 LangString str_msg_compat_title    ${LANG_SIMPCHINESE} \
-    " Vi Compatibility "
+    " Vi / Vim behavior "
+LangString str_msg_compat_desc     ${LANG_SIMPCHINESE} \
+    "&Compatibility and enhancements"
 LangString str_msg_compat_vi       ${LANG_SIMPCHINESE} \
-    " Vi compatible"
+    "Vi compatible"
 LangString str_msg_compat_vim      ${LANG_SIMPCHINESE} \
-    " Vim original"
+    "Vim original"
 LangString str_msg_compat_defaults ${LANG_SIMPCHINESE} \
-    " Some enhancements"
+    "Vim with some enhancements"
 LangString str_msg_compat_all      ${LANG_SIMPCHINESE} \
-    " All enhancements (Default)"
+    "Vim with all enhancements (Default)"
 
 LangString str_msg_keymap_title   ${LANG_SIMPCHINESE} \
-    " Key remapping for Windows behavior "
+    " Mappings "
+LangString str_msg_keymap_desc    ${LANG_SIMPCHINESE} \
+    "&Remap a few keys (Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F, etc)"
 LangString str_msg_keymap_default ${LANG_SIMPCHINESE} \
     " Do not remap keys (Default)"
 LangString str_msg_keymap_windows ${LANG_SIMPCHINESE} \
-    " Remap a few keys (<C-V>, <C-C>, <C-A>, <C-S>, <C-F>, etc)"
+    " Remap a few keys"
 
 LangString str_msg_mouse_title   ${LANG_SIMPCHINESE} \
-    " Mouse behavior "
+    " Mouse "
+LangString str_msg_mouse_desc    ${LANG_SIMPCHINESE} \
+    "&Behavior of right and left buttons"
 LangString str_msg_mouse_default ${LANG_SIMPCHINESE} \
-    " Default:    Right button has a popup menu, left button starts visual mode"
+    "Right: popup menu, Left: visual mode (Default)"
 LangString str_msg_mouse_windows ${LANG_SIMPCHINESE} \
-    " Windows: Right button has a popup menu, left button starts select mode"
+    "Right: popup menu, Left: select mode (Windows)"
 LangString str_msg_mouse_unix    ${LANG_SIMPCHINESE} \
-    " Unix:        Right button extends selection, left button starts visual mode"
+    "Right: extends selection, Left: visual mode (Unix)"
diff --git a/nsis/lang/tradchinese.nsi b/nsis/lang/tradchinese.nsi
index 14471c19a..aff28edf3 100644
--- a/nsis/lang/tradchinese.nsi
+++ b/nsis/lang/tradchinese.nsi
@@ -226,31 +226,37 @@ LangString str_msg_rm_root_fail  ${LANG_TRADCHINESE} \
 LangString str_vimrc_page_title    ${LANG_TRADCHINESE} \
     "Choose _vimrc settings"
 LangString str_vimrc_page_subtitle ${LANG_TRADCHINESE} \
-    "Choose the default _vimrc settings."
+    "Choose the settings for enhancement, keyboard and mouse."
 
 LangString str_msg_compat_title    ${LANG_TRADCHINESE} \
-    " Vi Compatibility "
+    " Vi / Vim behavior "
+LangString str_msg_compat_desc     ${LANG_TRADCHINESE} \
+    "&Compatibility and enhancements"
 LangString str_msg_compat_vi       ${LANG_TRADCHINESE} \
-    " Vi compatible"
+    "Vi compatible"
 LangString str_msg_compat_vim      ${LANG_TRADCHINESE} \
-    " Vim original"
+    "Vim original"
 LangString str_msg_compat_defaults ${LANG_TRADCHINESE} \
-    " Some enhancements"
+    "Vim with some enhancements"
 LangString str_msg_compat_all      ${LANG_TRADCHINESE} \
-    " All enhancements (Default)"
+    "Vim with all enhancements (Default)"
 
 LangString str_msg_keymap_title   ${LANG_TRADCHINESE} \
-    " Key remapping for Windows behavior "
+    " Mappings "
+LangString str_msg_keymap_desc    ${LANG_TRADCHINESE} \
+    "&Remap a few keys (Ctrl-V, Ctrl-C, Ctrl-A, Ctrl-S, Ctrl-F, etc)"
 LangString str_msg_keymap_default ${LANG_TRADCHINESE} \
     " Do not remap keys (Default)"
 LangString str_msg_keymap_windows ${LANG_TRADCHINESE} \
-    " Remap a few keys (<C-V>, <C-C>, <C-A>, <C-S>, <C-F>, etc)"
+    " Remap a few keys"
 
 LangString str_msg_mouse_title   ${LANG_TRADCHINESE} \
-    " Mouse behavior "
+    " Mouse "
+LangString str_msg_mouse_desc    ${LANG_TRADCHINESE} \
+    "&Behavior of right and left buttons"
 LangString str_msg_mouse_default ${LANG_TRADCHINESE} \
-    " Default:    Right button has a popup menu, left button starts visual mode"
+    "Right: popup menu, Left: visual mode (Default)"
 LangString str_msg_mouse_windows ${LANG_TRADCHINESE} \
-    " Windows: Right button has a popup menu, left button starts select mode"
+    "Right: popup menu, Left: select mode (Windows)"
 LangString str_msg_mouse_unix    ${LANG_TRADCHINESE} \
-    " Unix:        Right button extends selection, left button starts visual mode"
+    "Right: extends selection, Left: visual mode (Unix)"
-- 
2.17.0

